File: excel/rhino/update/excel_update.js
#! /usr/bin/java org.mozilla.javascript.tools.shell.Main
// ----------------------------------------------------------------
/*
	excel_update.js
					Aug/08/2013

*/
// ----------------------------------------------------------------
load ("/var/www/data_base/common/rhino_common/text_manipulate.js");
load ("/var/www/data_base/common/rhino_common/excel_manipulate.js");

// ----------------------------------------------------------------
print ("*** 開始 ***");

var excel_file = arguments[0];
var key=arguments[1];
var population_in= parseInt (arguments[2]);

print	("key = " + key + "\tpopulation_in = " + population_in);

var dict_aa = excel_read_proc (excel_file);

if (key in dict_aa)
	{
	var dict_bb = dict_update_proc (dict_aa,key,population_in);
	dict_display_proc (dict_bb);
	excel_write_proc (excel_file,dict_bb);
	}

print ("*** 終了 ***");  
  
// ----------------------------------------------------------------

File: excel/python/update/excel_update.py
#! /usr/bin/python
# -*- coding: utf-8 -*-
#
#	update/excel_update.py
#
#					Oct/16/2016
import	sys
import	string
#
# ---------------------------------------------------------------
sys.path.append ('/var/www/data_base/common/python_common')
from text_manipulate import dict_update_proc
from excel_manipulate import excel_read_proc
from excel_manipulate import excel_write_proc
# ---------------------------------------------------------------
#
sys.stderr.write ("*** 開始 ***\n")
#
excel_file = sys.argv[1]
id_in = sys.argv[2]
population_in = int (sys.argv[3])
print ("%s\t%d" % (id_in, population_in))

dict_aa = excel_read_proc	(excel_file)

dict_bb = dict_update_proc (dict_aa,id_in,population_in)

excel_write_proc (excel_file,dict_bb)
#
sys.stderr.write ("*** 終了 ***\n")
# ---------------------------------------------------------------
